tests/dataset-hash-collisions/README.md

```diff
@@ -0,0 +1,21 @@
+Test Description
+================
+
+Datasets use a static DJB2 hash function to hash all types of datasets. These hashed
+datasets are stored in the THash API which has no randomization in place. As
+a result of this, the hash table can be exploited with a worst case time scenario of
+O(n) where n is the total number of entries in the table as a result of excessive chaining
+in a single row.
+
+The test shows that it takes excess time for the THash API to load the datasets from the file
+as many of them evaluate the exact same hash using the algorithm so this is not even the worst
+case scenario. With bigger dataset and lesser system specs/availability of resources,
+this can be worse. Note that it is not just about the number of datasets as there already
+does exist a test already that loads 1m+ datasets.
```

Comment:
Could we do with less than a million entries ?

Reply:
sounds enough

Reply:
Please note that Suricata does exit gracefully w 64k strings. Takes about 2m13s on my system w debug flags enabled. 😉

Reply:
Do you mean 2m13s with or without the patch ?

Reply:
without the patch 😉